### PR TITLE
Obj_Iceblock

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: include/variables.h [hunk 1]
```diff
@@ -4102,6 +4102,8 @@ extern Gfx D_050219E0[];
 extern UNK_TYPE D_05023008;
 extern UNK_TYPE D_0502324C;
 
+extern CollisionHeader D_06000438;
+
```
[the remark this answers is not in the document]
Comment (Kenix3) [reply include/variables.h:4106]: +1 to this

**Review by Kenix3 — CHANGES_REQUESTED** [review]

File: include/functions.h [hunk 2]
```diff
@@ -665,8 +665,8 @@ void func_800B3030(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f*
 // void EffectSsDeadDs_Spawn(UNK_TYPE4 uParm1, Vec3f* pzParm2, Vec3f* pzParm3, Vec3f* pzParm4, UNK_TYPE2 param_5, UNK_TYPE2 param_6, UNK_TYPE2 param_7, UNK_TYPE4 param_8);
 // void func_800B31BC(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE2 param_5, UNK_TYPE4 param_6);
 void EffectSsIceSmoke_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale);
-// void EffectSsIceBlock_Spawn(UNK_TYPE4 uParm1, Vec3f* pzParm2, Vec3f* pzParm3, Vec3f* pzParm4, UNK_TYPE2 param_5);
-void FlagSet_Update(GameState* gameState);
+void EffectSsIceBlock_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale);
+void FlagSet_Update(GameState* gamestate);
```
Comment (AngheloAlf) [inline include/functions.h:669]:
```suggestion
void FlagSet_Update(GameState* gameState);
```

File: src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c [hunk 3]
```diff
@@ -28,158 +62,1538 @@ const ActorInit Obj_Iceblock_InitVars = {
...
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+
+    if (OBJICEBLOCK_GET_1(&this->dyna.actor)) {
+        this->collider.dim.yShift = -100;
+        this->collider.dim.height = 126;
+        this->collider.dim.radius = 29;
+    }
+
+    if (D_80A26E7C == NULL) {
+        D_80A26E7C = (AnimatedMaterial*)Lib_SegmentedToVirtual(&D_06000328);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c:971]: I think there's no need for this cast
```suggestion
        D_80A26E7C = Lib_SegmentedToVirtual(&D_06000328);
```

File: src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c [hunk 4]
```diff
@@ -28,158 +62,1538 @@ const ActorInit Obj_Iceblock_InitVars = {
...
+    this->actionFunc = func_80A266E0;
+    this->unk_2A8 = 0.0f;
+}
+
+void func_80A266E0(ObjIceblock* this, GlobalContext* globalCtx) {
+    f32 temp;
+    s32 sp20;
+    Actor* actor = &this->dyna.actor;
+
+    Math_StepToF(&this->unk_2A8, (1.0f / 600.0f), 0.00016666666f);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c:1429]: Would this match?
```suggestion
    Math_StepToF(&this->unk_2A8, (1.0f / 600.0f), (1.0f / 6000.0f));
```

File: src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c [hunk 5]
```diff
@@ -28,158 +62,1538 @@ const ActorInit Obj_Iceblock_InitVars = {
...
+
+s32 func_80A243E0(ObjIceblock* this, GlobalContext* globalCtx, Vec3f* arg0) {
+    static f32 D_80A26F30[] = {
+        -300.0f,
+        300.0f,
+    };
+    static f32 D_80A26F38[] = {
+        1.0f,
+        -1.0f,
+    };
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c:527]: Same here, does removing the trailing `,`s result in single lines?

File: src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c [hunk 6]
```diff
@@ -28,158 +62,1538 @@ const ActorInit Obj_Iceblock_InitVars = {
...
+    if (this) {}
+
+    if (sp30) {
+        func_80A2541C(this, globalCtx);
+    }
+    func_80A24A48(this, globalCtx);
+}
+
+void func_80A260E8(ObjIceblock* this) {
+    static f32 D_80A26FC0[] = {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Iceblock/z_obj_iceblock.c:1258]: Same question here.

**Review by hensldm — COMMENTED** [review]
A few formatting questions.

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by hensldm — CHANGES_REQUESTED** [review]
With the object now, don't forget to remove the symbols from `undefined_syms.txt`.

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**